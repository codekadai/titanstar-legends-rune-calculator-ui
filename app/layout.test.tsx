import React from "react";
import { expect, it, vi } from "vitest";
import { render } from "@testing-library/react";
import RootLayout from "./layout";
import Page from "./(with-layout)/page";

vi.mock("next/font/google", () => ({
  Montserrat: () => ({
    style: {
      fontFamily: "mocked",
    },
  }),
}));

it("renders correctly", () => {
  const rootLayout = render(
    <RootLayout>
      <Page />
    </RootLayout>
  );
  expect(rootLayout).toMatchSnapshot();
});
