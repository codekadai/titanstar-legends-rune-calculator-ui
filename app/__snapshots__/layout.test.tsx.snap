// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <html
        lang="en"
      >
        <body>
          <section
            class="_container_a35809"
          >
            <div
              class="_treeWrapper_068057"
            >
              <div
                class="_loaderWrapper_459d5b"
              >
                <p
                  class="_loaderText_459d5b"
                >
                  Please wait while we fetch your stats...
                </p>
                <div
                  class="_loader_459d5b"
                />
              </div>
            </div>
          </section>
        </body>
      </html>
    </div>
  </body>,
  "container": <div>
    <html
      lang="en"
    >
      <body>
        <section
          class="_container_a35809"
        >
          <div
            class="_treeWrapper_068057"
          >
            <div
              class="_loaderWrapper_459d5b"
            >
              <p
                class="_loaderText_459d5b"
              >
                Please wait while we fetch your stats...
              </p>
              <div
                class="_loader_459d5b"
              />
            </div>
          </div>
        </section>
      </body>
    </html>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
